src/webgpu/shader/execution/robust_access_vertex.spec.ts

```diff
@@ -44,7 +44,430 @@ Vertex buffer contents could be randomized to prevent the case where a previous
+      .filter(
+        ({ indexed, drawCallTestParameter }) =>
+          (indexed && drawCallTestParameter in IndexedDrawCallParameter) ||
+          (!indexed && drawCallTestParameter in NonIndexedDrawCallParameter)
+      )
+      .combine(poptions('type', Object.keys(typeInfoMap)))
+      .combine(poptions('additionalBuffers', [0, 4]))
+      .combine(poptions('errorScale', [1, 4, 10 ** 2, 10 ** 4, 10 ** 6]))
+  )
+  .fn(async t => {
+    const p = t.params;
+    const typeInfo = typeInfoMap[p.type];
+
+    // Number of vertices to draw
+    const numVertices = 3;
+    // Each buffer will be bound to this many slots (2 would mean 2 attributes per buffer)
+    const slotsPerBuffer = 2;
+    // Make an array big enough for the vertices, slots, and size of each element
+    const vertexArray = new Float32Array(numVertices * slotsPerBuffer * typeInfo.size);
+
+    // Sufficiently unusual values to fill our buffer with to avoid collisions with other tests
+    const arbitraryValues = [759, 329, 908];
+    for (let i = 0; i < vertexArray.length; ++i) {
+      vertexArray[i] = arbitraryValues[i % arbitraryValues.length];
+    }
+    // A valid value is 0 or one in the buffer
+    const validValues = [0, ...arbitraryValues];
+
+    // Instance step mode buffer, vertex step mode buffer
+    const bufferContents = [vertexArray, vertexArray];
+    // Additional buffers (vertex step mode)
+    for (let i = 0; i < p.additionalBuffers; i++) {
+      bufferContents.push(vertexArray);
+    }
+
+    // Mutable draw call
+    const draw = new DrawCall(t.device, bufferContents, numVertices, slotsPerBuffer);
+
+    // Create attributes listing
+    let layoutStr = '';
+    const attributeNames = [];
+    {
+      let currSlot = 0;
+      for (let i = 0; i < bufferContents.length; i++) {
+        for (let j = 0; j < slotsPerBuffer; j++) {
+          layoutStr += `layout(location=${currSlot}) in ${p.type} a_${currSlot};\n`;
+          attributeNames.push(`a_${currSlot}`);
+          currSlot++;
+        }
+      }
+    }
+
+    // Vertex buffer descriptors
+    const vertexBuffers: GPUVertexBufferLayoutDescriptor[] = [];
+    {
+      let currSlot = 0;
+      for (let i = 0; i < bufferContents.length; i++) {
+        vertexBuffers.push({
+          arrayStride: slotsPerBuffer * typeInfo.size,
+          stepMode: i === 0 ? 'instance' : 'vertex',
+          attributes: Array(slotsPerBuffer)
+            .fill(0)
+            .map((_, i) => ({
+              shaderLocation: currSlot++,
+              offset: i * typeInfo.size,
+              format: typeInfo.format,
+            })),
+        });
+      }
+    }
+
+    // Offset the range checks for gl_VertexIndex in the shader if we use BaseVertex
+    let vertexIndexOffset = 0;
+    if (p.drawCallTestParameter === DrawCallParameter.BaseVertex) {
+      vertexIndexOffset += p.errorScale;
+    }
+
+    // Construct pipeline that outputs a green fragment, only if we notice any invalid values
```

Comment:
nit: I would reverse the green/red semantic so that green is good and red is bad

Reply:
Done